### ISSUE TITLE
DOS mouse usage crashes keyboard/mouse input

### ISSUE THREAD

**Issue by alexoughton** [MiSTer-devel/ao486_MiSTer#64]
Hello,

I am trying to use ao486 with DOS 5 and Windows 3.1, and finding that loading a DOS mouse driver will cause all keyboard input to stop working. The menu can still be triggered and controlled with WIN+F12, but ao486 will not respond to any more keyboard/mouse input until a reset is performed via CTRL+ALT+ALT.

Steps to reproduce:

1. Boot up ao486 (no mouse driver loaded)
2. Start Windows. Mouse is fine. Exit Windows. (This Windows test step can be skipped with no effect. Just proves things are otherwise working).
3. Load DOS mouse driver. Can be CuteMouse or Microsoft driver I have, with the same behavior.
4. Don't touch the mouse. Everything is fine. Keyboard is responsive.
5. Move mouse. Cursor does not move. Keyboard has stopped responding.

I'm not seeing anything I recognize in the console output (over UART). In case it's important, the following appears upon starting the mouse driver:

mouse_ctl = 0xF2
mouse_reply = 0xFA
mouse_reply = 0x00
mouse_ctl = 0xF2
mouse_reply = 0xFA
mouse_reply = 0x00
mouse_ctl = 0xF4
mouse_reply = 0xFA
mouse_ctl = 0xF3
mouse_reply = 0xFA
mouse_ctl = 0x64
mouse_reply = 0xFA

The following appears upon starting a DOS game (in this case, Sierra's Space Quest 4):

mouse_ctl = 0xF4
mouse_reply = 0xFA
mouse_ctl = 0xF3
mouse_reply = 0xFA
mouse_ctl = 0x64
mouse_reply = 0xFA

Nothing appears when I move the mouse to trigger the freeze.

The keyboard/mouse combination I am using are a wireless combo on a shared Logitech USB receiver. Unplugging and replugging the receiver does not seem to have an effect. Plugging in a different keyboard (also via a logitech receiver - all I have right now) also seems to have no effect.

Any thoughts here? Any way I can get some more diagnostic information in the console so I can find out what might be going wrong?

Thanks in advance,

Alex Oughton

**Comment by sorgelig** [comment]
to summarize above: mouse driver doesn't work for you?
There are 2 possible issues:
1) himem/xmem manager issue. Boot without memory manager and then try mouse driver.
2) try other version of ctmouse - some versions may not work well.

**Comment by alexoughton** [comment]
Hi sorgelig,

Thanks for responding. To summarize the above, using the mouse in DOS crashes both mouse and keyboard input such that neither will work any more until reset.

I have tried disabling himem.sys and emm386 and the problem remains. I don't even have to launch a game, just load the mouse driver and move the mouse. Keyboard input is now dead. I have also tried the recommended settings from this forum post:

https://misterfpga.org/viewtopic.php?f=13&t=955&p=8051&hilit=mouse#p8051

Same effect. Any move of the mouse crashes both mouse and keyboard. I have also just upgraded from DOS 5 to 6.22, and no improvement. Also, I have tried several versions of ctmouse as well as the Microsoft mouse driver and the same problem occurs each time.

**Comment by alexoughton** [comment]
Also, I have just tried limiting the RAM size from 256MB to 16MB, with no effect.

**Comment by sorgelig** [comment]
I cannot reproduce your problem, alas. Mouse simply works for me.

**Comment by alexoughton** [comment]
I've just completed erased my mister, started again with mister-fusion and installed ao486 with DOS 6.22 from scratch. Now the mouse is working correctly in DOS.

I made a backup image of my original mister and will do a side-by-side comparison of all settings. Clearly I changed something in my original attempt which causes this lock-up. If I find the setting which causes it, I will report back.

**Comment by alexoughton** [comment]
I've got the bottom of this. Incredibly, this was sound-related.

The problem was caused by the sound driver I was using for my DOS game (Space Quest 4). I was experiencing stuttering/jumbled audio when using the "Sound Blaster" driver, and had switched to the "Thunder Board" driver as this solved a very similar problem on my real 486 PC (with an SB16). On ao486 however, this solves nothing and causes the keyboard/mouse problem.

The reason I didn't initially make this connection is because the problem would persist across a reset (CTRL+ALT+ALT). If I had _ever_ used the Thunder Board driver since power on, then the mouse problem would occur even if I never loaded the game again after the reset. This made it look like the problem had nothing to do with my software. However, I eventually realized that the problem would go away if I did a full power-cycle and switched back to the "Sound Blaster" driver. The stuttering sound problem was solved by using sbctl.exe and SET BLASTER to use SBPRO-compatible mode instead of SB16.
